pcdsdevices/ccm.py

```diff
@@ -270,33 +286,25 @@ def alio_to_theta(alio, theta0, gr, gd):
     docstring, either to indicate your success or to increment the hours
     counter below.
 
+    Conversion function
+    theta_angle = f(x) = 2arctan * [(sqrt(x^2 + D^2 + 2Rx) - D)/(2R + x)]
+    Note that for x = −R, θ = 2 arctan(−R/D)
+
     total hours spent here: 2
     """
-
-    low = -1.0
-    high = 1.0
-    timeout = 1.0
-    start = time.time()
-    while time.time() - start < timeout:
-        theta_guess = (low+high)/2
-        alio_calc = theta_to_alio(theta_guess, theta0, gr, gd)
-        if np.isclose(alio, alio_calc):
-            break
-        elif alio_calc > alio:
-            high = theta_guess
-        else:
-            low = theta_guess
-    return theta_guess
+    return theta0 + 180 / np.pi * 2 * np.arctan(
+        (np.sqrt(alio ** 2 + gd ** 2 + 2 * gr * alio) - gd) / (2 * gr + alio)
+    )
 
 
 def wavelength_to_theta(wavelength, dspacing):
     """Converts wavelength (A) to theta angle (rad)."""
-    return np.arcsin(wavelength/2/dspacing)
+    return 180.0 / np.pi * np.arcsin(wavelength / 2 / dspacing)
```

Comment:
OK this is a weird one here..... when I implemented the old function back for `alio_to_theta` and `theta_to_alio` - these two functions: `wavelength_to_theta` and `theta_to_wavelenght` were failing the tests.... usually we would probably change the tests, but for curiosity I changed these functions to use the `old` implementations, mainly added the conversion from radians to degrees back and it seems to work fine now... 
Simply putting back the `default_theta0 = 14.9792 * np.pi/180` was not fixing the problem - it has to do with the calculation of `theta` as well....
Can anybody please give me a bit of insight here ?? 
Was this part of the new requirements?

Reply:
I wrote these tests to verify that the calculations were reversible. The old calculations are not reversible (which is why I updated them). Therefore, the tests fail.

Reply:
I suggest just putting an xfail on this test

Reply:
It is not really that they're failing the assertion, it is more like failing the calculation: 
` ValueError: Position is outside of pseudo single limits: fake_ccm_calc_energy, 4 < -5.02697921205642 < 25`
```
FAILED tests/test_ccm.py::test_ccm_calc - ValueError: Position is outside of pseudo single limits: fake_ccm_calc_energy, 4 < -5.02697921205642 < 25
FAILED tests/test_ccm.py::test_vernier - ValueError: Position is outside of pseudo single limits: fake_ccm_calc_energy, 4 < -5.02697921205642 < 25
```

I need to look into it more...

Reply:
Ok I had some conversion issues - that's why I was getting these (thanks @klauer for pointing that our for me on slack)

But.... wasn't it supposed to fail??? Am I still doing something wrong?? the test seems to work fine ... maybe i should wait until passes the checks though

Reply:
It may just be that there's one value that is reversible here? Not sure, but it seems a good thing anyway.

Reply:
On the note of testing, it might not be a bad idea to use the old code and pick a few sets of calculation results that we can keep as regression test cases.

For example, if we were testing `cos(x)` we could say `test_cases = [(0, 1), (np.pi / 2, 0), ...]``

Reply:
@klauer that would be great - where in the old code I can I get a few sets of calculation results from? - I am looking the `old-hutch-configs` repo, but can't find these - I might be looking for the wrong things though